BUG: wrongly assumed Path when parsing html

I see this in one of the astroquery tests (reproduce it with `pytest -P mpc -R`, I may write up an MRE without referring to astroquery later):
 It's clearly caused by 3bf33f68ad220a00384d42478e43955f62454afe, the tests pass with 6.1.7
```

self = <astroquery.mpc.tests.test_mpc_remote.TestMPC object at 0x136b73740>

    def test_get_observations(self):
        # asteroids
        a = mpc.MPC.get_observations(2)
        assert a['number'][0] == 2
    
        a = mpc.MPC.get_observations(12893)
        assert a['number'][0] == 12893
        assert a['desig'][-1] == '1998 QS55'
>       a = mpc.MPC.get_observations("2019 AA")

astroquery/mpc/tests/test_mpc_remote.py:97: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
astroquery/utils/class_or_instance.py:25: in f
    return self.fn(obj, *args, **kwds)
astroquery/utils/process_asyncs.py:29: in newmethod
    result = self._parse_result(response, verbose=verbose)
astroquery/mpc/core.py:1212: in _parse_result
    data = ascii.read("\n".join([o['original_record']
../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/io/ascii/ui.py:398: in read
    new_kwargs["guess_html"] = _probably_html(table)
../../../.pyenv/versions/3.12.1/lib/python3.12/site-packages/astropy/io/ascii/ui.py:148: in _probably_html
    table_path = Path(table).expanduser()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = PosixPath('~0FR1K19A00A  C2011 01 29.24643 01 18 02.537-02 41 30.21         22.2 wL~3JL8F51\n~0FR1K19A00A  C2011 01 29...47 46 56.60         20.93GV~7ukZG96\n~0FR1K19A00A 1C2024 03 03.20377105 56 18.827+47 46 54.95         20.97GV~7ukZG96')

    def expanduser(self):
        """ Return a new path with expanded ~ and ~user constructs
        (as returned by os.path.expanduser)
        """
        if (not (self.drive or self.root) and
            self._tail and self._tail[0][:1] == '~'):
            homedir = self._flavour.expanduser(self._tail[0])
            if homedir[:1] == "~":
>               raise RuntimeError("Could not determine home directory.")
E               RuntimeError: Could not determine home directory.

../../../.pyenv/versions/3.12.1/lib/python3.12/pathlib.py:1406: RuntimeError
==================================================================== short test summary info ====================================================================
FAILED astroquery/mpc/tests/test_mpc_remote.py::TestMPC::test_get_observations - RuntimeError: Could not determine home directory.

```

--- a/astropy/io/ascii/tests/test_read.py
+++ b/astropy/io/ascii/tests/test_read.py
@@ -1497,6 +1497,13 @@ def test_probably_html(home_is_data):
         ],
         (" <! doctype htm > ", " hello world"),
         [[1, 2, 3]],
+        # regression tests for https://github.com/astropy/astropy/issues/17562
+        "~itsatrap",  # looks like a Path, but isn't
+        (
+            "~0FR1K19A00A  C2011 01 29.24643 01 18 02.537-02 41 30.21         22.2 wL~3JL8F51\n"
+            "~0FR1K19A00A  C2011 01 29...47 46 56.60         20.93GV~7ukZG96\n"
+            "~0FR1K19A00A 1C2024 03 03.20377105 56 18.827+47 46 54.95         20.97GV~7ukZG96"
+        ),
     ):
         assert _probably_html(tabl0) is False
 

--- a/astropy/io/ascii/ui.py
+++ b/astropy/io/ascii/ui.py
@@ -145,8 +145,10 @@ def _probably_html(table, maxchars=100000):
             return True
 
         # Filename ending in .htm or .html which exists
-        table_path = Path(table).expanduser()
-        if re.search(r"\.htm[l]?$", table[-5:], re.IGNORECASE) and table_path.is_file():
+        if (
+            re.search(r"\.htm[l]?$", table[-5:], re.IGNORECASE)
+            and Path(table).expanduser().is_file()
+        ):
             return True
 
         # Table starts with HTML document type declaration

--- a/docs/changes/io.ascii/17565.bugfix.rst
+++ b/docs/changes/io.ascii/17565.bugfix.rst
@@ -1,0 +1,1 @@
+Fixed parsing ASCII table with data that starts with a tilda.

